TombEngine/Game/missile.cpp

```diff
@@ -151,35 +151,36 @@ void ControlNatlaGun(short fxNumber)
 			return;
 		}
 
-		fxNumber = CreateNewEffect(pointColl.RoomNumber);
-		if (fxNumber != NO_ITEM)
-		{
-			auto& fxNew = EffectList[fxNumber];
-
-			fxNew.pos.Position = pointColl.Coordinates;
-			fxNew.pos.Orientation.y = fx.pos.Orientation.y;
-			fxNew.roomNumber = pointColl.RoomNumber;
-			fxNew.speed = fx.speed;
-			fxNew.frameNumber = 0;
-			fxNew.objectNumber = ID_PROJ_BOMB;
-		}
+		fxNumber = CreateNewEffect(pointColl.RoomNumber, ID_PROJ_BOMB, fx.Pose.Position);
+		if (fxNumber == NO_ITEM)
+			return;
+
+		auto& fxNew = g_Level.Items[fxNumber];
+
+		fxNew.Pose.Position = pointColl.Coordinates;
+		fxNew.Pose.Orientation.y = fx.Pose.Orientation.y;
+		fxNew.RoomNumber = pointColl.RoomNumber;
+		fxNew.Animation.Velocity.z = fx.Animation.Velocity.z;
+		fxNew.Animation.FrameNumber = 0;
+		fxNew.ObjectNumber = ID_PROJ_BOMB;
 	}
 }
 
 short ShardGun(int x, int y, int z, short velocity, short yRot, short roomNumber)
 {
-	int fxNumber = CreateNewEffect(roomNumber);
+	int fxNumber = CreateNewEffect(roomNumber, ID_PROJ_SHARD, Pose(x, y, z));
 	if (fxNumber != NO_ITEM)
 	{
-		auto& fx = EffectList[fxNumber];
-
-		fx.pos.Position = Vector3i(x, y, z);
-		fx.pos.Orientation = EulerAngles(0, yRot, 0);
-		fx.roomNumber = roomNumber;
-		fx.speed = velocity;
-		fx.frameNumber = 0;
-		fx.objectNumber = ID_PROJ_SHARD;
-		fx.color = Vector4::One;
+		auto& fx = g_Level.Items[fxNumber];
+
+		fx.Pose.Position = Vector3i(x, y, z);
+		fx.Pose.Orientation = EulerAngles(0, yRot, 0);
+		fx.RoomNumber = roomNumber;
+		fx.Animation.Velocity.z = velocity;
+		fx.Animation.FrameNumber = 0;
+		fx.ObjectNumber = ID_PROJ_SHARD;
```

Comment:
Do we need to re-assign object number if it was already done in CreateNewEffect above?